gqcp/include/FockSpace/Configuration.hpp

```diff
@@ -30,6 +30,14 @@ namespace GQCP {
 struct Configuration {
```

Comment:
I'm not really sure `Configuration` is the correct name for this class. In the light of the CI refactor (also related to #345), what do you think of `SpinResolvedONV`, where the `SpinResolved` part means loosely 'has both alpha and beta spin'?